agroal/src/main/java/org/wildfly/extension/agroal/service/DataSourceService.java

```diff
@@ -0,0 +1,136 @@
+        this.dataSourceName = dataSourceName;
+        this.jndiName = jndiName;
+        this.connectable = connectable;
+        this.xa = xa;
+        this.dataSourceConfiguration = dataSourceConfiguration;
+    }
+
+    @Override
+    public void start(StartContext context) throws StartException {
+        Class<?> providerClass = driverService.getValue().providerClass();
+        if ( xa ) {
+            if ( !XADataSource.class.isAssignableFrom( providerClass ) ) {
+                throw new StartException( "An xa-datasource requires a javax.sqlXADataSource as connection provider. Fix the connection-provider for the driver" );
+            }
+        } else {
+            if ( providerClass != null && !DataSource.class.isAssignableFrom( providerClass ) && !Driver.class.isAssignableFrom( providerClass ) ) {
+                throw new StartException( "Invalid connection provider. Either a java.sql.Driver or javax.sql.DataSource implementation is required. Fix the connection-provider for the driver" );
+            }
+        }
+
+        dataSourceConfiguration.connectionPoolConfiguration( cp -> cp.connectionFactoryConfiguration( cf -> cf.connectionProviderClass( providerClass ) ) );
+
+        if ( transactionManager.getOptionalValue() != null && transactionSynchronizationRegistry.getOptionalValue() != null ) {
+            NarayanaTransactionIntegration txIntegration = new NarayanaTransactionIntegration( transactionManager.getValue(), transactionSynchronizationRegistry.getValue(), jndiName, connectable );
+            dataSourceConfiguration.connectionPoolConfiguration( cp -> cp.transactionIntegration( txIntegration ) );
+        } else if ( xa ) {
+            throw new StartException( "Could not start xa-datasource: transaction manager is missing" );
+        }
+
+        try {
+            agroalDataSource = AgroalDataSource.from( dataSourceConfiguration, new LoggingDataSourceListener( dataSourceName ) );
+
+            ContextNames.BindInfo bindInfo = ContextNames.bindInfoFor( jndiName );
+            BinderService binderService = new BinderService( bindInfo.getBindName() );
+            ImmediateManagedReferenceFactory managedReferenceFactory = new ImmediateManagedReferenceFactory( agroalDataSource );
+            context.getChildTarget().addService( bindInfo.getBinderServiceName(), binderService )
+                    .addInjectionValue( binderService.getManagedObjectInjector(), new ImmediateValue<ManagedReferenceFactory>( managedReferenceFactory ) )
+                    .addDependency( bindInfo.getParentContextServiceName(), ServiceBasedNamingStore.class, binderService.getNamingStoreInjector() )
+                    .install();
+
+            AgroalLogger.SERVICE_LOGGER.infof( "Started %sdatasource '%s' bound to [%s]", xa ? "xa-" : "", dataSourceName, jndiName );
+        } catch ( SQLException e ) {
+            agroalDataSource = null;
+            throw new StartException( "Exception starting datasource " + dataSourceName, e );
```

Comment:
ok, I'll mention it one last time :)  Any exception or log above info must be i18ned